Bootstrap service

aqus opened the pull request.

Commit:
impl

Commit:
remove streaming

Commit:
checkstyle

Commit:
actuator

Commit:
transactional save file

Commit:
SessionCreationPolicy.STATELESS

stvort commented on src/main/java/com/tasktrace/actuator/ApplicationHealthIndicator.java:
```diff
@@ -0,0 +1,13 @@
+public class ApplicationHealthIndicator implements HealthIndicator {
+    @Override
+    public Health health() {
+        return Health.up().build();
```
Зачем это нужно?

stvort commented on src/main/java/com/tasktrace/controller/FileController.java:
```diff
@@ -0,0 +1,42 @@
+        return ResponseEntity.ok(fileUuid);
+    }
+
+    @GetMapping("file/{uuid}")
```
`api/v1/file/{uuid}`?

stvort commented on src/main/java/com/tasktrace/controller/FileController.java:
```diff
@@ -0,0 +1,42 @@
+
+    @GetMapping("file/{uuid}")
+    public ResponseEntity<byte[]> readFile(@PathVariable UUID uuid) {
+        FileServiceImpl.ChunkWithMetadata chunkWithMetadata = fileService.fetchFile(uuid);
```
И так мы привязали контроллер к конкретной реализации сервиса файлов

stvort commented on src/main/java/com/tasktrace/service/FileServiceImpl.java:
```diff
@@ -0,0 +1,63 @@
+    @Override
+    public ChunkWithMetadata fetchFile(UUID uuid) {
+        FileMetadataEntity fileMetadata = fileMetadataRepository.findById(uuid.toString())
+                .orElseThrow(() -> new EntityNotFoundException("File with id %s is not found".formatted(uuid)));
```
Причем тут `EntityNotFoundException`? Разве файл это доменная сущность?

aqus committed:
Update README to reflect changes in content and structure

aqus committed:
Update endpoint mapping in `FileController` to use API versioning prefix.

aqus committed:
Refactor file service interfaces and implementations to use a unified model for file metadata.

This commit updates the `FileService` interface and its implementation to use a more generic model (`ChunkWithMetadata`) that includes both metadata and the actual file chunk data. This change simplifies the interaction between controllers and services, making it clearer what is being returned and facilitating future modifications or extensions of this functionality.

aqus committed:
Add a descriptive class name and method signature for the newly created `FileNotFoundException` class. Update related imports and usages of `EntityNotFoundException` to reference `FileNotFoundException`.

aqus committed:
Remove unnecessary import statement.
